From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Wed, 24 Jun 2020 21:49:22 +0200
Subject: [PATCH 183/194] pine64_defconfig: cleanup defconfig and enable
 additional drivers

For Mobian, many drivers and features (such as ACPI, EFI or PCI support)
are useless and can be safely removed.

In order to improve usability and performance, we also add a number of
drivers/features:
- general: full preemption, ZRAM
- security: AppArmor
- networking: Bluetooth, Netfilter, Wireguard
- filesystems: F2FS, NTFS, ExFAT, CIFS, dm-crypt, FS-based encryption
- media: PinePhone/PineTab cameras & flash LED controller, cedrus
- USB: gadget, type-C, ANX7688
- Android/anbox: binder, ashmem
- graphics (including backlight) drivers built-in for early display init
---
 arch/arm64/configs/pine64_defconfig | 234 ++++++++++++------------------------
 1 file changed, 74 insertions(+), 160 deletions(-)

diff --git a/arch/arm64/configs/pine64_defconfig b/arch/arm64/configs/pine64_defconfig
index 472f010..feafeaa 100644
--- a/arch/arm64/configs/pine64_defconfig
+++ b/arch/arm64/configs/pine64_defconfig
@@ -38,7 +38,6 @@ CONFIG_SLAB_FREELIST_RANDOM=y
 CONFIG_SLAB_FREELIST_HARDENED=y
 CONFIG_PROFILING=y
 CONFIG_ARCH_SUNXI=y
-CONFIG_ARCH_ROCKCHIP=y
 # CONFIG_ARM64_ERRATUM_832075 is not set
 # CONFIG_ARM64_ERRATUM_834220 is not set
 # CONFIG_ARM64_ERRATUM_1024718 is not set
@@ -85,10 +84,9 @@ CONFIG_CP15_BARRIER_EMULATION=y
 # CONFIG_ARM64_E0PD is not set
 # CONFIG_ARCH_RANDOM is not set
 CONFIG_RANDOMIZE_BASE=y
+# CONFIG_EFI is not set
 CONFIG_PM_AUTOSLEEP=y
 CONFIG_PM_WAKELOCKS=y
-CONFIG_PM_DEBUG=y
-CONFIG_PM_TEST_SUSPEND=y
 CONFIG_WQ_POWER_EFFICIENT_DEFAULT=y
 CONFIG_ENERGY_MODEL=y
 CONFIG_CPU_IDLE=y
@@ -108,7 +106,6 @@ CONFIG_ARM_SCPI_CPUFREQ=m
 CONFIG_ARM_SCPI_PROTOCOL=m
 CONFIG_ARM_SDE_INTERFACE=y
 CONFIG_FIRMWARE_MEMMAP=y
-CONFIG_DMI_SYSFS=m
 CONFIG_VIRTUALIZATION=y
 CONFIG_KVM=y
 CONFIG_CRYPTO_SHA256_ARM64=y
@@ -123,7 +120,6 @@ CONFIG_CRYPTO_AES_ARM64_CE_BLK=m
 CONFIG_CRYPTO_NHPOLY1305_NEON=m
 CONFIG_CRYPTO_AES_ARM64_BS=m
 CONFIG_JUMP_LABEL=y
-# CONFIG_GCC_PLUGINS is not set
 CONFIG_MODULES=y
 CONFIG_MODULE_FORCE_LOAD=y
 CONFIG_MODULE_UNLOAD=y
@@ -168,6 +164,13 @@ CONFIG_NETFILTER=y
 CONFIG_NF_CONNTRACK=m
 # CONFIG_NF_CONNTRACK_PROCFS is not set
 CONFIG_NF_CONNTRACK_EVENTS=y
+CONFIG_NF_CONNTRACK_FTP=m
+CONFIG_NF_CONNTRACK_H323=m
+CONFIG_NF_CONNTRACK_IRC=m
+CONFIG_NF_CONNTRACK_SNMP=m
+CONFIG_NF_CONNTRACK_PPTP=m
+CONFIG_NF_CONNTRACK_SIP=m
+CONFIG_NF_CONNTRACK_TFTP=m
 CONFIG_NF_TABLES=y
 CONFIG_NF_TABLES_INET=y
 CONFIG_NF_TABLES_NETDEV=y
@@ -192,10 +195,27 @@ CONFIG_NFT_TPROXY=m
 CONFIG_NFT_SYNPROXY=m
 CONFIG_NFT_DUP_NETDEV=m
 CONFIG_NFT_FWD_NETDEV=m
+CONFIG_NETFILTER_XT_MARK=m
 CONFIG_NETFILTER_XT_TARGET_CHECKSUM=m
+CONFIG_NETFILTER_XT_TARGET_CONNMARK=m
 CONFIG_NETFILTER_XT_TARGET_LOG=m
 CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=m
+CONFIG_NETFILTER_XT_MATCH_COMMENT=m
+CONFIG_NETFILTER_XT_MATCH_CONNLABEL=m
+CONFIG_NETFILTER_XT_MATCH_CONNLIMIT=m
+CONFIG_NETFILTER_XT_MATCH_CONNMARK=m
 CONFIG_NETFILTER_XT_MATCH_CONNTRACK=m
+CONFIG_NETFILTER_XT_MATCH_DEVGROUP=m
+CONFIG_NETFILTER_XT_MATCH_LENGTH=m
+CONFIG_NETFILTER_XT_MATCH_LIMIT=m
+CONFIG_NETFILTER_XT_MATCH_MAC=m
+CONFIG_NETFILTER_XT_MATCH_MULTIPORT=m
+CONFIG_NETFILTER_XT_MATCH_NFACCT=m
+CONFIG_NETFILTER_XT_MATCH_OSF=m
+CONFIG_NETFILTER_XT_MATCH_OWNER=m
+CONFIG_NETFILTER_XT_MATCH_QUOTA=m
+CONFIG_NETFILTER_XT_MATCH_SOCKET=m
+CONFIG_NETFILTER_XT_MATCH_STATE=m
 CONFIG_NFT_DUP_IPV4=m
 CONFIG_NFT_FIB_IPV4=m
 CONFIG_NF_TABLES_ARP=y
@@ -204,6 +224,7 @@ CONFIG_IP_NF_FILTER=m
 CONFIG_IP_NF_TARGET_REJECT=m
 CONFIG_IP_NF_NAT=m
 CONFIG_IP_NF_TARGET_MASQUERADE=m
+CONFIG_IP_NF_TARGET_REDIRECT=m
 CONFIG_IP_NF_MANGLE=m
 CONFIG_NFT_DUP_IPV6=m
 CONFIG_NFT_FIB_IPV6=m
@@ -237,6 +258,7 @@ CONFIG_NET_CLS_ACT=y
 CONFIG_NET_ACT_GACT=m
 CONFIG_NET_ACT_MIRRED=m
 CONFIG_NET_ACT_GATE=m
+CONFIG_VSOCKETS=m
 CONFIG_CGROUP_NET_PRIO=y
 CONFIG_CGROUP_NET_CLASSID=y
 CONFIG_BPF_JIT=y
@@ -259,42 +281,19 @@ CONFIG_MAC80211=m
 CONFIG_RFKILL=m
 CONFIG_RFKILL_INPUT=y
 CONFIG_RFKILL_GPIO=m
-CONFIG_PCI=y
-CONFIG_PCIEPORTBUS=y
-CONFIG_PCIEAER=y
-CONFIG_PCIE_ECRC=y
-CONFIG_PCIEASPM_POWER_SUPERSAVE=y
-CONFIG_PCI_STUB=m
-CONFIG_PCI_HOST_GENERIC=m
-CONFIG_PCIE_ROCKCHIP_HOST=m
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 CONFIG_FW_LOADER_COMPRESS=y
 # CONFIG_ALLOW_DEV_COREDUMP is not set
-CONFIG_MTD=y
-CONFIG_MTD_CMDLINE_PARTS=y
-CONFIG_MTD_BLOCK=y
-CONFIG_MTD_OOPS=m
-CONFIG_MTD_PARTITIONED_MASTER=y
-CONFIG_MTD_CFI=y
-CONFIG_MTD_JEDECPROBE=y
-CONFIG_MTD_CFI_ADV_OPTIONS=y
-CONFIG_MTD_CFI_INTELEXT=y
-CONFIG_MTD_CFI_AMDSTD=y
-CONFIG_MTD_CFI_STAA=y
-CONFIG_MTD_SPI_NOR=y
 CONFIG_OF_OVERLAY=y
 CONFIG_OF_CONFIGFS=y
 CONFIG_ZRAM=m
+CONFIG_ZRAM_WRITEBACK=y
+CONFIG_ZRAM_MEMORY_TRACKING=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_NBD=m
-CONFIG_ATA_OVER_ETH=m
-CONFIG_BLK_DEV_NVME=m
 CONFIG_DUMMY_IRQ=m
 CONFIG_SRAM=y
-CONFIG_EEPROM_AT24=m
-CONFIG_EEPROM_AT25=m
-CONFIG_EEPROM_93CX6=m
 CONFIG_SENSORS_LIS3_I2C=m
 CONFIG_SCSI=y
 # CONFIG_SCSI_PROC_FS is not set
@@ -305,7 +304,6 @@ CONFIG_SCSI_CONSTANTS=y
 CONFIG_SCSI_SCAN_ASYNC=y
 # CONFIG_SCSI_LOWLEVEL is not set
 CONFIG_ATA=m
-CONFIG_SATA_AHCI=m
 CONFIG_SATA_AHCI_PLATFORM=m
 # CONFIG_ATA_SFF is not set
 CONFIG_MD=y
@@ -335,28 +333,17 @@ CONFIG_NETCONSOLE=m
 CONFIG_NETCONSOLE_DYNAMIC=y
 CONFIG_TUN=m
 CONFIG_VETH=m
-# CONFIG_NET_VENDOR_3COM is not set
-# CONFIG_NET_VENDOR_ADAPTEC is not set
-# CONFIG_NET_VENDOR_AGERE is not set
 # CONFIG_NET_VENDOR_ALACRITECH is not set
 # CONFIG_NET_VENDOR_ALLWINNER is not set
-# CONFIG_NET_VENDOR_ALTEON is not set
 # CONFIG_NET_VENDOR_AMAZON is not set
 # CONFIG_NET_VENDOR_AMD is not set
 # CONFIG_NET_VENDOR_AQUANTIA is not set
 # CONFIG_NET_VENDOR_ARC is not set
-# CONFIG_NET_VENDOR_ATHEROS is not set
 # CONFIG_NET_VENDOR_AURORA is not set
 # CONFIG_NET_VENDOR_BROADCOM is not set
-# CONFIG_NET_VENDOR_BROCADE is not set
 # CONFIG_NET_VENDOR_CADENCE is not set
 # CONFIG_NET_VENDOR_CAVIUM is not set
-# CONFIG_NET_VENDOR_CHELSIO is not set
-# CONFIG_NET_VENDOR_CISCO is not set
 # CONFIG_NET_VENDOR_CORTINA is not set
-# CONFIG_NET_VENDOR_DEC is not set
-# CONFIG_NET_VENDOR_DLINK is not set
-# CONFIG_NET_VENDOR_EMULEX is not set
 # CONFIG_NET_VENDOR_EZCHIP is not set
 # CONFIG_NET_VENDOR_GOOGLE is not set
 # CONFIG_NET_VENDOR_HISILICON is not set
@@ -367,42 +354,28 @@ CONFIG_VETH=m
 # CONFIG_NET_VENDOR_MICREL is not set
 # CONFIG_NET_VENDOR_MICROCHIP is not set
 # CONFIG_NET_VENDOR_MICROSEMI is not set
-# CONFIG_NET_VENDOR_MYRI is not set
 # CONFIG_NET_VENDOR_NATSEMI is not set
-# CONFIG_NET_VENDOR_NETERION is not set
 # CONFIG_NET_VENDOR_NETRONOME is not set
 # CONFIG_NET_VENDOR_NI is not set
-# CONFIG_NET_VENDOR_NVIDIA is not set
-# CONFIG_NET_VENDOR_OKI is not set
-# CONFIG_NET_VENDOR_PACKET_ENGINES is not set
 # CONFIG_NET_VENDOR_PENSANDO is not set
-# CONFIG_NET_VENDOR_QLOGIC is not set
 # CONFIG_NET_VENDOR_QUALCOMM is not set
-# CONFIG_NET_VENDOR_RDC is not set
-# CONFIG_NET_VENDOR_REALTEK is not set
 # CONFIG_NET_VENDOR_RENESAS is not set
 # CONFIG_NET_VENDOR_ROCKER is not set
 # CONFIG_NET_VENDOR_SAMSUNG is not set
 # CONFIG_NET_VENDOR_SEEQ is not set
 # CONFIG_NET_VENDOR_SOLARFLARE is not set
-# CONFIG_NET_VENDOR_SILAN is not set
-# CONFIG_NET_VENDOR_SIS is not set
 # CONFIG_NET_VENDOR_SMSC is not set
 # CONFIG_NET_VENDOR_SOCIONEXT is not set
 CONFIG_STMMAC_ETH=m
 # CONFIG_DWMAC_GENERIC is not set
 # CONFIG_DWMAC_SUNXI is not set
-# CONFIG_NET_VENDOR_SUN is not set
 # CONFIG_NET_VENDOR_SYNOPSYS is not set
-# CONFIG_NET_VENDOR_TEHUTI is not set
-# CONFIG_NET_VENDOR_TI is not set
 # CONFIG_NET_VENDOR_VIA is not set
 # CONFIG_NET_VENDOR_WIZNET is not set
 # CONFIG_NET_VENDOR_XILINX is not set
 CONFIG_MDIO_DEVICE=y
 CONFIG_LED_TRIGGER_PHY=y
 CONFIG_REALTEK_PHY=m
-CONFIG_ROCKCHIP_PHY=m
 CONFIG_USB_NET_DRIVERS=m
 CONFIG_USB_PEGASUS=m
 CONFIG_USB_RTL8150=m
@@ -416,21 +389,19 @@ CONFIG_USB_NET_SMSC75XX=m
 CONFIG_USB_NET_SMSC95XX=m
 CONFIG_USB_NET_PLUSB=m
 CONFIG_USB_NET_MCS7830=m
+CONFIG_USB_NET_RNDIS_HOST=m
+# CONFIG_USB_ARMLINUX is not set
+# CONFIG_USB_NET_ZAURUS is not set
 CONFIG_USB_NET_QMI_WWAN=m
 # CONFIG_WLAN_VENDOR_ADMTEK is not set
 CONFIG_ATH9K=m
 CONFIG_ATH10K=m
-CONFIG_ATH10K_PCI=m
 CONFIG_ATH10K_SDIO=m
 # CONFIG_WLAN_VENDOR_ATMEL is not set
 CONFIG_BRCMFMAC=m
 # CONFIG_WLAN_VENDOR_CISCO is not set
 # CONFIG_WLAN_VENDOR_INTEL is not set
 # CONFIG_WLAN_VENDOR_INTERSIL is not set
-CONFIG_MWIFIEX=m
-CONFIG_MWIFIEX_SDIO=m
-CONFIG_MWIFIEX_PCIE=m
-CONFIG_MWIFIEX_USB=m
 # CONFIG_WLAN_VENDOR_MEDIATEK is not set
 # CONFIG_WLAN_VENDOR_MICROCHIP is not set
 # CONFIG_WLAN_VENDOR_RALINK is not set
@@ -442,6 +413,7 @@ CONFIG_RTL8XXXU=m
 # CONFIG_WLAN_VENDOR_ZYDAS is not set
 # CONFIG_WLAN_VENDOR_QUANTENNA is not set
 CONFIG_INPUT_FF_MEMLESS=y
+CONFIG_INPUT_POLLDEV=m
 CONFIG_INPUT_EVDEV=y
 CONFIG_KEYBOARD_ADC=m
 # CONFIG_KEYBOARD_ATKBD is not set
@@ -469,7 +441,6 @@ CONFIG_INPUT_PWM_VIBRA=m
 CONFIG_SERIAL_8250=y
 # CONFIG_SERIAL_8250_DEPRECATED_OPTIONS is not set
 CONFIG_SERIAL_8250_CONSOLE=y
-# CONFIG_SERIAL_8250_PCI is not set
 CONFIG_SERIAL_8250_EXTENDED=y
 CONFIG_SERIAL_8250_SHARE_IRQ=y
 CONFIG_SERIAL_8250_DW=y
@@ -477,11 +448,8 @@ CONFIG_SERIAL_OF_PLATFORM=y
 CONFIG_NULL_TTY=m
 CONFIG_SERIAL_DEV_BUS=y
 CONFIG_TTY_PRINTK=m
-# CONFIG_HW_RANDOM_CAVIUM is not set
-# CONFIG_DEVPORT is not set
 CONFIG_TCG_TPM=m
 CONFIG_TCG_FTPM_TEE=m
-CONFIG_I2C=y
 CONFIG_I2C_CHARDEV=y
 CONFIG_I2C_MUX=y
 CONFIG_I2C_MUX_GPIO=m
@@ -489,20 +457,12 @@ CONFIG_I2C_MUX_PINCTRL=m
 CONFIG_I2C_GPIO=m
 CONFIG_I2C_MV64XXX=y
 CONFIG_I2C_RK3X=y
-CONFIG_SPI=y
-CONFIG_SPI_GPIO=m
-CONFIG_SPI_ROCKCHIP=y
-CONFIG_SPI_SUN6I=y
-CONFIG_SPI_SPIDEV=y
 # CONFIG_PTP_1588_CLOCK is not set
 CONFIG_PINCTRL_AXP209=y
 CONFIG_GPIO_SYSFS=y
-CONFIG_GPIO_MAX732X=y
-CONFIG_GPIO_MAX732X_IRQ=y
 CONFIG_GPIO_PCA953X=y
 CONFIG_GPIO_PCA953X_IRQ=y
 CONFIG_POWER_AVS=y
-CONFIG_ROCKCHIP_IODOMAIN=y
 CONFIG_POWER_RESET_GPIO=y
 CONFIG_POWER_RESET_GPIO_RESTART=y
 CONFIG_POWER_RESET_SYSCON=y
@@ -513,6 +473,7 @@ CONFIG_BATTERY_CW2015=m
 CONFIG_CHARGER_AXP20X=m
 CONFIG_BATTERY_AXP20X=m
 CONFIG_AXP20X_POWER=m
+CONFIG_AXP288_FUEL_GAUGE=m
 CONFIG_SENSORS_ARM_SCPI=m
 CONFIG_SENSORS_DRIVETEMP=m
 CONFIG_SENSORS_GPIO_FAN=m
@@ -531,7 +492,6 @@ CONFIG_DEVFREQ_THERMAL=y
 CONFIG_THERMAL_EMULATION=y
 CONFIG_THERMAL_MMIO=m
 CONFIG_SUN8I_THERMAL=y
-CONFIG_ROCKCHIP_THERMAL=y
 CONFIG_GENERIC_ADC_THERMAL=m
 CONFIG_WATCHDOG=y
 CONFIG_WATCHDOG_SYSFS=y
@@ -540,7 +500,6 @@ CONFIG_SUNXI_WATCHDOG=y
 CONFIG_ARM_SMC_WATCHDOG=y
 CONFIG_MFD_AXP20X_I2C=y
 CONFIG_MFD_AXP20X_RSB=y
-CONFIG_MFD_RK808=y
 CONFIG_MFD_SUN6I_PRCM=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
@@ -550,14 +509,7 @@ CONFIG_REGULATOR_AXP20X=y
 CONFIG_REGULATOR_FAN53555=y
 CONFIG_REGULATOR_GPIO=y
 CONFIG_REGULATOR_PWM=y
-CONFIG_REGULATOR_RK808=y
-CONFIG_REGULATOR_SY8106A=y
 CONFIG_REGULATOR_VCTRL=y
-CONFIG_RC_CORE=m
-CONFIG_RC_DECODERS=y
-CONFIG_RC_DEVICES=y
-CONFIG_IR_SUNXI=m
-CONFIG_MEDIA_CEC_RC=y
 CONFIG_MEDIA_SUPPORT=m
 CONFIG_MEDIA_SUPPORT_FILTER=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
@@ -568,51 +520,24 @@ CONFIG_USB_VIDEO_CLASS=m
 CONFIG_V4L_PLATFORM_DRIVERS=y
 CONFIG_VIDEO_SUN6I_CSI=m
 CONFIG_V4L_MEM2MEM_DRIVERS=y
-CONFIG_VIDEO_ROCKCHIP_RGA=m
 CONFIG_VIDEO_SUN8I_DEINTERLACE=m
 CONFIG_VIDEO_SUN8I_ROTATE=m
-# CONFIG_VIDEO_IR_I2C is not set
 CONFIG_VIDEO_OV5640=m
 CONFIG_VIDEO_GC2145=m
-# CONFIG_VGA_ARB is not set
-CONFIG_DRM=m
-CONFIG_DRM_VGEM=m
-CONFIG_DRM_ROCKCHIP=m
-CONFIG_ROCKCHIP_ANALOGIX_DP=y
-CONFIG_ROCKCHIP_CDN_DP=y
-CONFIG_ROCKCHIP_DW_HDMI=y
-CONFIG_ROCKCHIP_DW_MIPI_DSI=y
-CONFIG_ROCKCHIP_INNO_HDMI=y
-CONFIG_ROCKCHIP_LVDS=y
-CONFIG_ROCKCHIP_RGB=y
-CONFIG_DRM_SUN4I=m
-# CONFIG_DRM_SUN4I_HDMI is not set
-# CONFIG_DRM_SUN4I_BACKEND is not set
-CONFIG_DRM_SUN6I_DSI=m
-CONFIG_DRM_SUN8I_DW_HDMI=m
-CONFIG_DRM_SUN8I_MIXER=m
-CONFIG_DRM_PANEL_LVDS=m
-CONFIG_DRM_PANEL_SIMPLE=m
-CONFIG_DRM_PANEL_FEIXIN_K101_IM2BA02=m
-CONFIG_DRM_PANEL_FEIYANG_FY07024DI26A30D=m
-CONFIG_DRM_PANEL_PINEPHONE_JD9365DA=m
-CONFIG_DRM_PANEL_SITRONIX_ST7703=m
-CONFIG_DRM_DISPLAY_CONNECTOR=m
-CONFIG_DRM_LVDS_CODEC=m
-CONFIG_DRM_SIMPLE_BRIDGE=m
-CONFIG_DRM_ANALOGIX_ANX6345=m
-CONFIG_DRM_DW_HDMI_AHB_AUDIO=m
-CONFIG_DRM_DW_HDMI_I2S_AUDIO=m
-CONFIG_DRM_DW_HDMI_CEC=m
-CONFIG_DRM_LIMA=m
-CONFIG_DRM_PANFROST=m
-CONFIG_FB=y
-CONFIG_FB_EFI=y
+CONFIG_DRM=y
+CONFIG_DRM_SUN4I=y
+CONFIG_DRM_SUN6I_DSI=y
+CONFIG_DRM_SUN8I_DW_HDMI=y
+CONFIG_DRM_SUN8I_MIXER=y
+CONFIG_DRM_PANEL_SIMPLE=y
+CONFIG_DRM_PANEL_ILITEK_ILI9881C=y
+CONFIG_DRM_PANEL_FEIYANG_FY07024DI26A30D=y
+CONFIG_DRM_PANEL_PINEPHONE_JD9365DA=y
+CONFIG_DRM_PANEL_SITRONIX_ST7703=y
+CONFIG_DRM_LIMA=y
 CONFIG_FB_SIMPLE=y
-CONFIG_BACKLIGHT_CLASS_DEVICE=m
-CONFIG_BACKLIGHT_PWM=m
-CONFIG_BACKLIGHT_GPIO=m
-CONFIG_BACKLIGHT_LED=m
+CONFIG_BACKLIGHT_CLASS_DEVICE=y
+CONFIG_BACKLIGHT_PWM=y
 CONFIG_FRAMEBUFFER_CONSOLE=y
 CONFIG_FRAMEBUFFER_CONSOLE_ROTATION=y
 CONFIG_FRAMEBUFFER_CONSOLE_DEFERRED_TAKEOVER=y
@@ -622,15 +545,11 @@ CONFIG_SND_HRTIMER=m
 CONFIG_SND_DYNAMIC_MINORS=y
 CONFIG_SND_VERBOSE_PRINTK=y
 CONFIG_SND_DEBUG=y
+CONFIG_SND_SEQUENCER=m
+CONFIG_SND_SEQ_DUMMY=m
 # CONFIG_SND_DRIVERS is not set
-# CONFIG_SND_PCI is not set
-# CONFIG_SND_SPI is not set
 CONFIG_SND_USB_AUDIO=m
 CONFIG_SND_SOC=m
-CONFIG_SND_SOC_ROCKCHIP=m
-CONFIG_SND_SOC_ROCKCHIP_SPDIF=m
-CONFIG_SND_SOC_ROCKCHIP_RT5645=m
-CONFIG_SND_SOC_RK3399_GRU_SOUND=m
 CONFIG_SND_SUN4I_CODEC=m
 CONFIG_SND_SUN8I_CODEC=m
 CONFIG_SND_SUN8I_CODEC_ANALOG=m
@@ -639,7 +558,6 @@ CONFIG_SND_SUN4I_I2S=m
 CONFIG_SND_SUN4I_SPDIF=m
 CONFIG_SND_SOC_BT_SCO=m
 CONFIG_SND_SOC_EC25=m
-CONFIG_SND_SOC_ES8316=m
 CONFIG_SND_SOC_SIMPLE_AMPLIFIER=m
 CONFIG_SND_SOC_SPDIF=m
 CONFIG_SND_SIMPLE_CARD=m
@@ -651,6 +569,7 @@ CONFIG_HID_LOGITECH=m
 CONFIG_HID_LOGITECH_DJ=m
 CONFIG_HID_MAGICMOUSE=m
 CONFIG_HID_MICROSOFT=m
+CONFIG_HID_MULTITOUCH=y
 CONFIG_HID_SONY=m
 CONFIG_SONY_FF=y
 CONFIG_HID_WIIMOTE=m
@@ -679,7 +598,6 @@ CONFIG_USB_MUSB_HDRC=m
 CONFIG_USB_MUSB_SUNXI=m
 CONFIG_USB_DWC3=m
 CONFIG_USB_DWC3_ULPI=y
-# CONFIG_USB_DWC3_HAPS is not set
 CONFIG_USB_SERIAL=m
 CONFIG_USB_SERIAL_GENERIC=y
 CONFIG_USB_SERIAL_SIMPLE=m
@@ -707,6 +625,17 @@ CONFIG_USB_CONFIGFS_F_FS=y
 CONFIG_USB_CONFIGFS_F_UAC1=y
 CONFIG_USB_CONFIGFS_F_UAC2=y
 CONFIG_USB_CONFIGFS_F_HID=y
+CONFIG_USB_ETH=m
+CONFIG_USB_ETH_EEM=y
+CONFIG_USB_FUNCTIONFS=m
+CONFIG_USB_FUNCTIONFS_ETH=y
+CONFIG_USB_FUNCTIONFS_RNDIS=y
+CONFIG_USB_FUNCTIONFS_GENERIC=y
+CONFIG_USB_MASS_STORAGE=m
+CONFIG_USB_G_SERIAL=m
+CONFIG_USB_G_MULTI=m
+CONFIG_USB_G_MULTI_CDC=y
+CONFIG_USB_G_HID=m
 CONFIG_USB_G_DBGP=m
 CONFIG_TYPEC=m
 CONFIG_TYPEC_TCPM=m
@@ -720,13 +649,12 @@ CONFIG_MMC_BLOCK_MINORS=32
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_PLTFM=y
 CONFIG_MMC_SDHCI_OF_ARASAN=y
-CONFIG_MMC_SPI=y
 CONFIG_MMC_DW=y
-CONFIG_MMC_DW_ROCKCHIP=y
 CONFIG_MMC_SUNXI=y
 CONFIG_LEDS_CLASS=y
 CONFIG_LEDS_CLASS_FLASH=y
 CONFIG_LEDS_CLASS_MULTICOLOR=y
+CONFIG_LEDS_BRIGHTNESS_HW_CHANGED=y
 CONFIG_LEDS_GPIO=y
 CONFIG_LEDS_PWM=m
 CONFIG_LEDS_REGULATOR=m
@@ -736,7 +664,6 @@ CONFIG_LEDS_SGM3140=m
 CONFIG_LEDS_TRIGGER_TIMER=y
 CONFIG_LEDS_TRIGGER_ONESHOT=y
 CONFIG_LEDS_TRIGGER_DISK=y
-CONFIG_LEDS_TRIGGER_MTD=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=y
 CONFIG_LEDS_TRIGGER_BACKLIGHT=y
 CONFIG_LEDS_TRIGGER_CPU=y
@@ -750,12 +677,9 @@ CONFIG_LEDS_TRIGGER_NETDEV=y
 CONFIG_LEDS_TRIGGER_PATTERN=y
 CONFIG_LEDS_TRIGGER_AUDIO=y
 CONFIG_RTC_CLASS=y
-CONFIG_RTC_DRV_RK808=y
-CONFIG_RTC_DRV_PCF8563=y
-CONFIG_RTC_DRV_EFI=y
 CONFIG_RTC_DRV_SUN6I=y
 CONFIG_DMADEVICES=y
-CONFIG_DMA_SUN6I=m
+CONFIG_DMA_SUN6I=y
 CONFIG_PL330_DMA=m
 CONFIG_UDMABUF=y
 CONFIG_DMABUF_MOVE_NOTIFY=y
@@ -764,20 +688,17 @@ CONFIG_DMABUF_HEAPS_SYSTEM=y
 CONFIG_DMABUF_HEAPS_CMA=y
 # CONFIG_VIRTIO_MENU is not set
 CONFIG_VHOST_NET=m
+CONFIG_VHOST_VSOCK=m
 CONFIG_VHOST_CROSS_ENDIAN_LEGACY=y
 CONFIG_STAGING=y
 CONFIG_RTL8723BS=m
 CONFIG_R8712U=m
 CONFIG_R8188EU=m
 CONFIG_STAGING_MEDIA=y
-CONFIG_VIDEO_HANTRO=m
-CONFIG_VIDEO_ROCKCHIP_VDEC=m
 CONFIG_VIDEO_SUNXI=y
 CONFIG_VIDEO_SUNXI_CEDRUS=m
-CONFIG_PHY_ROCKCHIP_DPHY_RX0=m
-CONFIG_VIDEO_ROCKCHIP_ISP1=m
+CONFIG_ASHMEM=y
 CONFIG_RTL8723CS_NEW=m
-CONFIG_COMMON_CLK_RK808=y
 CONFIG_COMMON_CLK_SCPI=m
 CONFIG_COMMON_CLK_PWM=y
 # CONFIG_CLK_SUNXI is not set
@@ -786,23 +707,17 @@ CONFIG_HWSPINLOCK=y
 # CONFIG_HISILICON_ERRATUM_161010101 is not set
 # CONFIG_ARM64_ERRATUM_858921 is not set
 CONFIG_MAILBOX=y
-CONFIG_ROCKCHIP_MBOX=y
-CONFIG_ROCKCHIP_IOMMU=y
 CONFIG_SUN50I_IOMMU=y
-CONFIG_ROCKCHIP_PM_DOMAINS=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
 CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
 CONFIG_DEVFREQ_GOV_PASSIVE=y
-CONFIG_ARM_RK3399_DMC_DEVFREQ=m
 CONFIG_EXTCON_GPIO=m
 CONFIG_EXTCON_USB_GPIO=y
 CONFIG_IIO=m
 CONFIG_IIO_SW_DEVICE=m
 CONFIG_IIO_SW_TRIGGER=m
 CONFIG_AXP20X_ADC=m
-CONFIG_ROCKCHIP_SARADC=m
 CONFIG_INV_MPU6050_I2C=m
 CONFIG_STK3310=m
 CONFIG_IIO_ST_MAGN_3AXIS=m
@@ -811,17 +726,12 @@ CONFIG_IIO_INTERRUPT_TRIGGER=m
 CONFIG_IIO_TIGHTLOOP_TRIGGER=m
 CONFIG_IIO_SYSFS_TRIGGER=m
 CONFIG_PWM=y
-CONFIG_PWM_ROCKCHIP=y
 CONFIG_PWM_SUN4I=y
 CONFIG_PHY_SUN4I_USB=y
 CONFIG_PHY_SUN50I_USB3=y
-CONFIG_PHY_ROCKCHIP_EMMC=y
-CONFIG_PHY_ROCKCHIP_INNO_HDMI=m
-CONFIG_PHY_ROCKCHIP_INNO_USB2=y
-CONFIG_PHY_ROCKCHIP_PCIE=m
-CONFIG_PHY_ROCKCHIP_TYPEC=y
-CONFIG_ROCKCHIP_EFUSE=y
-CONFIG_ROCKCHIP_OTP=m
+CONFIG_ANDROID=y
+CONFIG_ANDROID_BINDER_IPC=y
+CONFIG_ANDROID_BINDERFS=y
 CONFIG_NVMEM_SUNXI_SID=y
 CONFIG_TEE=y
 CONFIG_OPTEE=y
@@ -841,6 +751,7 @@ CONFIG_BTRFS_FS=y
 CONFIG_BTRFS_FS_POSIX_ACL=y
 CONFIG_F2FS_FS=y
 CONFIG_F2FS_FS_SECURITY=y
+CONFIG_F2FS_FS_COMPRESSION=y
 # CONFIG_MANDATORY_FILE_LOCKING is not set
 CONFIG_FS_ENCRYPTION=y
 # CONFIG_DNOTIFY is not set
@@ -858,14 +769,16 @@ CONFIG_VFAT_FS=y
 CONFIG_FAT_DEFAULT_UTF8=y
 CONFIG_EXFAT_FS=m
 CONFIG_NTFS_FS=m
+CONFIG_NTFS_RW=y
 CONFIG_PROC_CHILDREN=y
 CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_TMPFS_INODE64=y
 CONFIG_HUGETLBFS=y
-CONFIG_EFIVAR_FS=y
 CONFIG_HFS_FS=m
 CONFIG_HFSPLUS_FS=m
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_XZ=y
 CONFIG_PSTORE=y
 CONFIG_PSTORE_PMSG=y
 CONFIG_PSTORE_RAM=y
@@ -884,6 +797,7 @@ CONFIG_NLS_UTF8=y
 CONFIG_SECURITY=y
 CONFIG_HARDENED_USERCOPY=y
 CONFIG_FORTIFY_SOURCE=y
+CONFIG_SECURITY_APPARMOR=y
 CONFIG_SECURITY_YAMA=y
 # CONFIG_INTEGRITY is not set
 CONFIG_INIT_ON_ALLOC_DEFAULT_ON=y
@@ -903,13 +817,13 @@ CONFIG_CRYPTO_USER_API_RNG=m
 CONFIG_CRYPTO_USER_API_AEAD=m
 CONFIG_CRYPTO_DEV_SUN8I_CE=m
 CONFIG_CRYPTO_DEV_SUN8I_CE_DEBUG=y
-CONFIG_CRYPTO_DEV_ROCKCHIP=m
 # CONFIG_RAID6_PQ_BENCHMARK is not set
+CONFIG_CRC_ITU_T=y
+CONFIG_CRC7=y
 CONFIG_DMA_CMA=y
-CONFIG_CMA_SIZE_MBYTES=32
+CONFIG_CMA_SIZE_MBYTES=256
 CONFIG_PRINTK_TIME=y
 CONFIG_BOOT_PRINTK_DELAY=y
-CONFIG_DYNAMIC_DEBUG=y
 CONFIG_MAGIC_SYSRQ=y
 # CONFIG_MAGIC_SYSRQ_SERIAL is not set
 CONFIG_DEBUG_FS=y
